digdag-tests/src/test/java/acceptance/RequireIT.java

```diff
@@ -425,4 +449,85 @@ private void prepareForChildWF(Path childOutFile)
             Files.write(projectDir.resolve("child.dig"), child.getBytes("UTF-8"));
         }
     }
+
+    @Test
+    public void testRequireHasTargetSessionIdEvenWhenJustWaiting()
+            throws Exception
+    {
+        // Create new project
+        CommandStatus initStatus = main("init",
+                "-c", config.toString(),
+                projectDir.toString());
+        assertThat(initStatus.errUtf8(), initStatus.code(), is(0));
+
+        Path childOutFile = projectDir.resolve("child.out").toAbsolutePath().normalize();
+        prepareForChildWF(childOutFile);
+        copyResource("acceptance/require/parent.dig", projectDir.resolve("parent.dig"));
+
+        // Push the project
+        CommandStatus pushStatus = main("push",
+                "--project", projectDir.toString(),
+                "require",
+                "-c", config.toString(),
+                "-e", server.endpoint(),
+                "-r", "4711");
+        assertThat(pushStatus.errUtf8(), pushStatus.code(), is(0));
+
+        // Start the child workflow first.
+        String session = "2016-01-01";
+
+        Id targetAttemptId;
+        Id targetSessionId;
+        {
+            CommandStatus startStatus = main("start",
+                    "-c", config.toString(),
+                    "-e", server.endpoint(),
+                    "require", "child",
+                    "--session", session);
+            assertThat(startStatus.code(), is(0));
+            targetAttemptId = getAttemptId(startStatus);
+            targetSessionId = getSessionId(startStatus);
+        }
+
+        // Start the workflow
+        Id attemptId;
+        {
+            CommandStatus startStatus = main("start",
+                    "-c", config.toString(),
+                    "-e", server.endpoint(),
+                    "require", "parent",
+                    "--session", session);
+            assertThat(startStatus.code(), is(0));
+            attemptId = getAttemptId(startStatus);
+        }
+
+        // Wait for the attempt to complete
+        boolean success = false;
+        for (int i = 0; i < 30; i++) {
```

Comment:
Increase the maximum wait to 60 may be better. We sometimes faces timeout because of CI performance.

Reply:
Thanks. Updated the limits. https://github.com/hkdnet/digdag/commit/f00f7002c6049bb42f6cac2a64dfec59a0f4665d